src/tbp/monty/frameworks/experiments/pretraining_experiments.py

```diff
@@ -67,21 +67,11 @@ def run_episode(self):
         objects.
         """
         self.pre_episode()
-        # set is_seeking_match False and goes in exploratory mode
-        self.model.step_type = "exploratory_step"
-        # Pass target info to model
-        target = self.dataloader.primary_target
-        self.model.detected_object = self.model.primary_target["object"]
-        for lm in self.model.learning_modules:
-            lm.detected_object = target["object"]
-            lm.buffer.stats["possible_matches"] = [target["object"]]
-            lm.buffer.stats["detected_location_on_model"] = (
-                self.first_epoch_object_location[target["object"]]
-            )
-            lm.buffer.stats["detected_location_rel_body"] = np.array(target["position"])
-            lm.buffer.stats["detected_rotation"] = target["euler_rotation"]
-            lm.detected_rotation_r = Rotation.from_quat(target["quat_rotation"]).inv()
-            lm.buffer.stats["detected_scale"] = target["scale"]
+        # Save compute if we are providing labels to all models, so don't need to
+        # perform matching parts of LM updates (default is matching_step)
+        if len(self.supervised_lm_ids) == len(self.model.learning_modules):
```

Comment:
I think in most cases this will be fine, but we're not actually confirming that these two lists have the same elements. I think the quickest way to do this would be to turn them both into sets and compare the sets.
```suggestion
        if set(self.supervised_lm_ids) == set(self.model.learning_modules):
```